pkg/certificate/mrc_reconciler.go

```diff
@@ -1,214 +1,212 @@
 package certificate
 
 import (
-	"time"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"fmt"
 
 	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
-	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/errcode"
 )
 
-// UseCase is how the certificate is used. Each use case will have its own rotation mechanism, that is necessary to
-// determine if we can propagate the rotation forward.
-type UseCase string
-
-const (
-	// UseCaseValidatingWebhook maps to the certificate use case for the validating webhook.
-	UseCaseValidatingWebhook UseCase = "validatingWebhook"
-
-	// UseCaseMutatingWebhook maps to the certificate use case for the Mutating Webhook
-	UseCaseMutatingWebhook UseCase = "mutatingWebhook"
-
-	// UseCaseXDSControlPlane maps to the certificate use case for the XDS Control Plane
-	UseCaseXDSControlPlane UseCase = "xdsControlPlane"
-
-	// UseCaseSidecar maps to the certificate use case for the  sidecar service certs
-	UseCaseSidecar UseCase = "sidecar"
+func (m *Manager) handleMRCEvent(event MRCEvent) error {
+	log.Debug().Msgf("handling MRC event for MRC %s", event.MRCName)
+	mrcList, err := m.mrcClient.ListMeshRootCertificates()
+	if err != nil {
+		return err
+	}
 
-	// UseCaseBootstrap maps to the certificate use case for the  Bootstrap secrets
-	UseCaseBootstrap UseCase = "bootstrap"
+	mrcListLen := len(mrcList)
+	if mrcListLen == 0 {
+		log.Error().Err(ErrMRCNotFound).Msgf("when handling MRC event for MRC %s, found no MRCs in OSM control plane namespace", event.MRCName)
+		return ErrMRCNotFound
+	}
+	if mrcListLen > 2 {
```

Comment:
I think we potentially could have more than 2, if some are in the `inactive` state

Reply:
if we filter inactive out then we don't need the additional checks for inactive in the `setissuers` below